constants.go

```diff
@@ -43,9 +45,9 @@ const (
     <graphics type='vnc' autoport='yes' listen='127.0.0.1'>
       <listen type='address' address='127.0.0.1'/>
     </graphics>
-    <interface type='network'>
+    <interface type='bridge'>
       <mac address='52:fd:fc:07:21:82'/>
-      <source network='{{.Network}}'/>
+      <source bridge='{{.Network}}'/>
```

Comment:
This is assuming that the bridge name used by crc is the same as the network name used by crc? I'd rather we avoid that. With the switch from `type='network'` to `type='bridge'`, how much of the code dealing with libvirt networks is still relevant?

Reply:
> This is assuming that the bridge name used by crc is the same as the network name used by crc?

Yes.



> I'd rather we avoid that.

Why?


> With the switch from `type='network'` to `type='bridge'`, how much of the code dealing with libvirt networks is still relevant?

I'm not sure I understand the question. The code here or in `crc`?

Reply:
> > This is assuming that the bridge name used by crc is the same as the network name used by crc?
> 
> Yes.
> 
> > I'd rather we avoid that.
> 
> Why?
> 

Because this will silently break if one day crc changed the libvirt network definition it creates:
```
<network>
  <name>crc</name>
...
  <bridge name='hastobecrc' stp='on' delay='0'/>
...
</network>
```
The network name is still 'crc', but the bridge name is now different ('hastobecrc')

Reply:
> > With the switch from `type='network'` to `type='bridge'`, how much of the code dealing with libvirt networks is still relevant?
> 
> I'm not sure I understand the question. The code here or in `crc`?

Reading https://libvirt.org/formatdomain.html#elementsNICS , I'm under the impression that with `<interface type='bridge'/>`, the existence of a libvirt network named `crc` is no longer required, all that matters is that there's a bridge with the correct name. So I feel it's a bit odd to have most of the machine driver code working with a libvirt crc network while we tell libvirt we only care about bridge name. Maybe it's just me :)

Reply:
correct me if wrong, but CRC connects the VM to a bridge? to identify this, it uses the bridge's name, right?

Reply:
> Because this will silently break if one day crc changed the libvirt network definition it creates:

Sure but:

1. That's not caused by this PR since the existing code assuming a network by the name 'crc' as well. The netwrok name can also change.
2. Hopefully we'll be moving this code into `crc` repo so then we can switch to using the same constant everywhere.

Reply:
> correct me if wrong, but CRC connects the VM to a bridge? to identify this, it uses the bridge's name, right?

Not sure if this was directed to me or @cfergeau but yes.

Reply:
> > Because this will silently break if one day crc changed the libvirt network definition it creates:
> 
> Sure but:
> 
>     1. That's not caused by this PR since the existing code assuming a network by the name 'crc' as well. The network name can also change.

The fact that we are hardcoding the network name to be 'crc' is not a good reason for now also hardcoding that the bridge name is the same as the network name. I'm not even sure the network name is 100% hardcoded because of GetCreateFlags().
Even if it's hardcoded, I don't see what prevents you from adding a DefaultBridge constant, or from parsing the network definition to get the bridge name instead of adding a not so obvious 
assumption on bridge name and network name.

Reply:
>  I don't see what prevents you from adding a DefaultBridge constant

That I surely can and should do. :)

> or from parsing the network definition to get the bridge name

That's more work and i don't see any advantage since I don't see why would we only decide to change the bridge name and not the network name. Having said that, I pushed a new patch to make this scenario further unlikely: https://github.com/code-ready/crc/pull/709/commits/703439193646115219a730ab3ad5da0c6c6a982e